### ISSUE TITLE
Reproduction error in Naive.ipynb - KeyError

### ISSUE THREAD

**Issue by zhangzw16** [SalesforceAIResearch/gift-eval#9]
I followed the tutorial to set up the environment and data, but encountered a bug in the _[4th code cell (Line 88)](https://github.com/SalesforceAIResearch/gift-eval/blob/main/notebooks/naive.ipynb)_ with the following error message. Switching to other datasets results in a similar **KeyError** bug.

```python3
        # Measure the time taken for evaluation
        res = evaluate_model(
            predictor,
            test_data=dataset.test_data,
            metrics=metrics,
            batch_size=512,
            axis=None,
            mask_invalid_label=True,
            allow_nan_forecast=False,
            seasonality=season_length,
        )
```

Error log:
```
{
	"name": "KeyError",
	"message": "'L2C_dim0_2021-02-08 00:00_0'",
	"stack": "---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[21], line 88
     79 predictor = NaivePredictor(
     80     dataset.prediction_length,
     81     season_length=season_length,
   (...)
     84     batch_size=512,
     85 )
     87 # Measure the time taken for evaluation
---> 88 res = evaluate_model(
     89     predictor,
     90     test_data=dataset.test_data,
     91     metrics=metrics,
     92     batch_size=512,
     93     axis=None,
     94     mask_invalid_label=True,
     95     allow_nan_forecast=False,
     96     seasonality=season_length,
     97 )
     99 # Append the results to the CSV file
    100 with open(csv_file_path, \"a\", newline=\"\") as csvfile:

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/gluonts/model/evaluation.py:262, in evaluate_model(model, test_data, metrics, axis, batch_size, mask_invalid_label, allow_nan_forecast, seasonality)
    247 \"\"\"
    248 Evaluate ``model`` when applied to ``test_data``, according to ``metrics``.
    249 
   (...)
    258 Return results as a Pandas ``DataFrame``.
    259 \"\"\"
    260 forecasts = model.predict(test_data.input)
--> 262 return evaluate_forecasts(
    263     forecasts=forecasts,
    264     test_data=test_data,
    265     metrics=metrics,
    266     axis=axis,
    267     batch_size=batch_size,
    268     mask_invalid_label=mask_invalid_label,
    269     allow_nan_forecast=allow_nan_forecast,
    270     seasonality=seasonality,
    271 )

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/gluonts/model/evaluation.py:207, in evaluate_forecasts(forecasts, test_data, metrics, axis, batch_size, mask_invalid_label, allow_nan_forecast, seasonality)
    182 def evaluate_forecasts(
    183     forecasts: Iterable[Forecast],
    184     *,
   (...)
    191     seasonality: Optional[int] = None,
    192 ) -> pd.DataFrame:
    193     \"\"\"
    194     Evaluate ``forecasts`` by comparing them with ``test_data``, according to
    195     ``metrics``.
   (...)
    205     Return results as a Pandas ``DataFrame``.
    206     \"\"\"
--> 207     metrics_values = evaluate_forecasts_raw(
    208         forecasts=forecasts,
    209         test_data=test_data,
    210         metrics=metrics,
    211         axis=axis,
    212         batch_size=batch_size,
    213         mask_invalid_label=mask_invalid_label,
    214         allow_nan_forecast=allow_nan_forecast,
    215         seasonality=seasonality,
    216     )
    217     index0 = metrics_values.pop(\"__index_0\", None)
    219     metric_shape = metrics_values[first(metrics_values)].shape

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/gluonts/model/evaluation.py:145, in evaluate_forecasts_raw(forecasts, test_data, metrics, axis, batch_size, mask_invalid_label, allow_nan_forecast, seasonality)
    142 forecast_batches = batcher(forecasts, batch_size=batch_size)
    144 pbar = tqdm()
--> 145 for input_batch, label_batch, forecast_batch in zip(
    146     input_batches, label_batches, forecast_batches
    147 ):
    148     if 0 not in axis:
    149         index_data.extend(
    150             [
    151                 (forecast.item_id, forecast.start_date)
    152                 for forecast in forecast_batch
    153             ]
    154         )

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/gluonts/itertools.py:128, in batcher.<locals>.get_batch()
    127 def get_batch():
--> 128     return list(itertools.islice(it, batch_size))

Cell In[20], line 160, in StatsForecastPredictor.predict(self, dataset, **kwargs)
    158     self.logger.info(f\"Processing final batch.\")
    159     results = self.sf_predict(pd.concat(batch.values()))
--> 160     yield from self.yield_forecast(batch.keys(), results)
    162 self.logger.info(\"Prediction completed.\")

Cell In[20], line 196, in StatsForecastPredictor.yield_forecast(self, item_ids, results)
    192 def yield_forecast(
    193     self, item_ids, results: pd.DataFrame
    194 ) -> Iterator[QuantileForecast]:
    195     for idx in item_ids:
--> 196         prediction = results.loc[idx]
    197         forecast_arrays = []
    198         model_name = self.ModelType.__name__

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/pandas/core/indexing.py:1103, in _LocationIndexer.__getitem__(self, key)
   1100 axis = self.axis or 0
   1102 maybe_callable = com.apply_if_callable(key, self.obj)
-> 1103 return self._getitem_axis(maybe_callable, axis=axis)

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/pandas/core/indexing.py:1343, in _LocIndexer._getitem_axis(self, key, axis)
   1341 # fall thru to straight lookup
   1342 self._validate_key(key, axis)
-> 1343 return self._get_label(key, axis=axis)

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/pandas/core/indexing.py:1293, in _LocIndexer._get_label(self, label, axis)
   1291 def _get_label(self, label, axis: AxisInt):
   1292     # GH#5567 this will fail if the label is not present in the axis.
-> 1293     return self.obj.xs(label, axis=axis)

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/pandas/core/generic.py:4095, in NDFrame.xs(self, key, axis, level, drop_level)
   4093             new_index = index[loc]
   4094 else:
-> 4095     loc = index.get_loc(key)
   4097     if isinstance(loc, np.ndarray):
   4098         if loc.dtype == np.bool_:

File ~/miniconda3/envs/gift-eval/lib/python3.10/site-packages/pandas/core/indexes/range.py:349, in RangeIndex.get_loc(self, key)
    347         raise KeyError(key) from err
    348 self._check_indexing_error(key)
--> 349 raise KeyError(key)

KeyError: 'L2C_dim0_2021-02-08 00:00_0'"
}
```

**Comment by cuthalionn** [comment]
Hi @zhangzw16 , this issue should now be resolved following the merge of [PR #6](https://github.com/SalesforceAIResearch/gift-eval/pull/6). Please give it another try and let us know if you’re still experiencing any problems!

**Comment by zhangzw16** [comment]
Thanks a lot. Everything is running smoothly now.
